collated/main/alexfoodw.md

```diff
@@ -0,0 +1,487 @@
+
+    private static String user;
+    private static String link;
+
+    /**
+     * Creates an AddCommand to add the specified {@code ReadOnlyPerson}
+     */
+    public FacebookLinkCommand(String url) {
+        link = url;
+    }
+
+    /**
+     * Completes the Facebook Link command
+     * @throws CommandException
+     */
+    public static void completeLink() throws CommandException {
+        Facebook facebookInstance = FacebookConnectCommand.getFacebookInstance();
+        user = null;
+        try {
+            user = facebookInstance.getName();
+            try {
+                facebookInstance.postLink(new URL(link));
+            } catch (MalformedURLException e) {
+                new Exception("Please enter a valid URL.");
+            }
+        } catch (FacebookException e) {
+            // exception not handled because Facebook API still throws an exception even if success,
+            // so exception is ignored for now
+            e.printStackTrace();
+        }
+
+        EventsCenter.getInstance().post(new NewResultAvailableEvent(
+                MESSAGE_FACEBOOK_LINK_SUCCESS + " (to " + user + "'s page.)", false));
+        BrowserPanel.setLink(false);
+    }
+
+    @Override
+    public CommandResult execute() throws CommandException {
+        if (!FacebookConnectCommand.isAuthenticated()) {
+            FacebookConnectCommand newFacebookConnect = new FacebookConnectCommand();
+            BrowserPanel.setLink(true);
+            newFacebookConnect.execute();
+            return new CommandResult(MESSAGE_FACEBOOK_LINK_INITIATED);
+        } else {
+            completeLink();
+            return new CommandResult(MESSAGE_FACEBOOK_LINK_SUCCESS + " (to " + user + "'s page.)");
+        }
+    }
+}
+```
+###### /main/java/seedu/address/logic/commands/FacebookPostCommand.java
+``` java
+/**
+ * Posts a message to a personal Facebook account.
+ */
+public class FacebookPostCommand extends Command {
+    public static final String COMMAND_WORD = "facebook post";
+    public static final String COMMAND_ALIAS = "fbpost";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": post a message to Facebook account\n"
+            + "Alias: " + COMMAND_ALIAS + "\n"
+            + "Parameters: MESSAGE\n"
+            + "Example: " + COMMAND_WORD + " hello world!";
+
+    public static final String MESSAGE_FACEBOOK_POST_SUCCESS = "Posted to Facebook!";
+    public static final String MESSAGE_FACEBOOK_POST_INITIATED = "User not authenticated, log in to proceed.";
+    public static final String MESSAGE_FACEBOOK_POST_ERROR = "Error posting to Facebook";
+
+    private static String user;
+    private static String toPost;
+
+    /**
+     * Creates an AddCommand to add the specified {@code ReadOnlyPerson}
+     */
+    public FacebookPostCommand(String message) {
+        toPost = message;
+    }
+
+    /**
+     * Completes the post command
+     * @throws CommandException
```

Comment:
Normally, you should give a description of the `@param` or the `@return` value. If you have nothing more to add, probably the comment line can be deleted as well (because it's not adding any value)?